urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/trash/HttpServer.java

```diff
@@ -0,0 +1,96 @@
+package trash;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+import websocket.common.Logger;
+
+// server runs single-threaded, sufficient for now because there is only one, simple request from each client
+// TODO this class needs some SERIOUS cleaning up to do
+/**
+ * @deprecated
+ */
+public class HttpServer implements Runnable
+{
+	private static final int PORT = 8080;
+	private boolean debug = false;
+
+	@Override
+	public void run()
+	{
+		try
+		{
+			// Create a ServerSocket to listen on that port.
+			final ServerSocket masterSocket = new ServerSocket(HttpServer.PORT);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *masterSocket*. Execution paths that do not contain closure statements were detected. Ensure *masterSocket* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).